[2025-07-19T14:20:51.496+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: build_ml_features.transform_features manual__2025-07-19T14:20:47.468554+00:00 [queued]>
[2025-07-19T14:20:51.499+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: build_ml_features.transform_features manual__2025-07-19T14:20:47.468554+00:00 [queued]>
[2025-07-19T14:20:51.499+0000] {taskinstance.py:2170} INFO - Starting attempt 1 of 2
[2025-07-19T14:20:51.503+0000] {taskinstance.py:2191} INFO - Executing <Task(_PythonDecoratedOperator): transform_features> on 2025-07-19 14:20:47.468554+00:00
[2025-07-19T14:20:51.506+0000] {standard_task_runner.py:60} INFO - Started process 126 to run task
[2025-07-19T14:20:51.508+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'build_ml_features', 'transform_features', 'manual__2025-07-19T14:20:47.468554+00:00', '--job-id', '110', '--raw', '--subdir', 'DAGS_FOLDER/build_ml_features_dag.py', '--cfg-path', '/tmp/tmpzolf53i6']
[2025-07-19T14:20:51.508+0000] {standard_task_runner.py:88} INFO - Job 110: Subtask transform_features
[2025-07-19T14:20:51.524+0000] {task_command.py:423} INFO - Running <TaskInstance: build_ml_features.transform_features manual__2025-07-19T14:20:47.468554+00:00 [running]> on host 36077f3335e6
[2025-07-19T14:20:51.549+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='data-platform' AIRFLOW_CTX_DAG_ID='build_ml_features' AIRFLOW_CTX_TASK_ID='transform_features' AIRFLOW_CTX_EXECUTION_DATE='2025-07-19T14:20:47.468554+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-19T14:20:47.468554+00:00'
[2025-07-19T14:20:51.568+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/decorators/base.py", line 241, in execute
    return_value = super().execute(context)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 199, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 216, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/build_ml_features_dag.py", line 79, in transform_features
    df.pipe(compute_ema)
  File "/home/airflow/.local/lib/python3.8/site-packages/polars/dataframe/frame.py", line 5762, in pipe
    return function(self, *args, **kwargs)
  File "/opt/airflow/features/features_v1.py", line 157, in compute_time_features
    pl.col("date").dt.is_month_end().cast(pl.Int8).alias("is_month_end"),
AttributeError: 'ExprDateTimeNameSpace' object has no attribute 'is_month_end'
[2025-07-19T14:20:51.572+0000] {taskinstance.py:1138} INFO - Marking task as UP_FOR_RETRY. dag_id=build_ml_features, task_id=transform_features, execution_date=20250719T142047, start_date=20250719T142051, end_date=20250719T142051
[2025-07-19T14:20:51.576+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 110 for task transform_features ('ExprDateTimeNameSpace' object has no attribute 'is_month_end'; 126)
[2025-07-19T14:20:51.607+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-07-19T14:20:51.615+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
